packages/front-end/components/Experiment/Templates/TemplateForm.tsx

```diff
@@ -0,0 +1,354 @@
+    }
+
+    const body = JSON.stringify(value);
+
+    const res = initialValue.id
+      ? await apiCall<{ template: ExperimentTemplateInterface }>(
+          `/templates/${initialValue.id}`,
+          {
+            method: "PUT",
+            body,
+          }
+        )
+      : await apiCall<{ template: ExperimentTemplateInterface }>("/templates", {
+          method: "POST",
+          body,
+        });
+
+    track("Create Experiment Template", {
+      source,
+      numTags: value.tags?.length || 0,
+      numMetrics:
+        (value.goalMetrics?.length || 0) +
+        (value.secondaryMetrics?.length || 0),
+    });
+
+    value.tags && refreshTags(value.tags);
+    mutateTemplates();
+    if (onCreate) {
+      onCreate(res.template.id);
+    } else {
+      router.push(`/experiments#templates`);
+    }
+  });
+
+  const availableProjects: SingleValue[] = projects
+    .slice()
+    .sort((a, b) => (a.name > b.name ? 1 : -1))
+    .filter((p) => permissionsUtils.canViewExperimentModal(p.id))
+    .map((p) => ({ value: p.id, label: p.name }));
+
+  const allowAllProjects = permissionsUtils.canViewExperimentModal();
+
+  const exposureQueryId = form.getValues("exposureQueryId");
+
+  const { currentProjectIsDemo } = useDemoDataSourceProject();
+
+  useEffect(() => {
+    const exposureQueries = datasource?.settings?.queries?.exposure || [];
+
+    if (!exposureQueries.find((q) => q.id === exposureQueryId)) {
+      form.setValue("exposureQueryId", exposureQueries?.[0]?.id ?? "");
+    }
+  }, [form, exposureQueryId, datasource?.settings?.queries?.exposure]);
+
+  let header = isNewTemplate
+    ? "Create Experiment Template"
+    : "Edit Experiment Template";
+  if (duplicate) {
+    header = "Duplicate Experiment Template";
+  }
+  const trackingEventModalType = kebabCase(header);
+
+  const nameFieldHandlers = form.register("templateMetadata.name", {
+    setValueAs: (s) => s?.trim(),
+  });
+
+  return (
+    <FormProvider {...form}>
+      <PagedModal
+        trackingEventModalType={trackingEventModalType}
+        trackingEventModalSource={source}
+        header={header}
+        close={onClose}
+        submit={onSubmit}
+        cta={"Save"}
+        ctaEnabled={canSubmit}
+        closeCta="Cancel"
+        size="lg"
+        step={step}
+        setStep={setStep}
+        backButton={true}
+        bodyClassName="px-4"
+        navFill
+      >
+        <Page display="Overview">
+          <div>
+            {msg && <div className="alert alert-info">{msg}</div>}
+
+            {currentProjectIsDemo && (
+              <div className="alert alert-warning">
+                You are creating a template under the demo datasource project.
+                This template will be deleted when the demo datasource project
+                is deleted.
+              </div>
+            )}
+
+            <h4 className="mb-3">Template Details</h4>
+
+            <Field
```

Comment:
can we apply a default name based on the experiment name?
`${experiment.name} Template` or something

Reply:
Done. Added name in `convertExperimentToTemplate`